torchgeo/datasets/south_america_soybean.py

```diff
@@ -0,0 +1,227 @@
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 after downloading files (may be slow)
+        Raises:
+            FileNotFoundError: if no files are found in ``root``
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+            AssertionError: if ``year`` is invalid
+        """
+        assert set(years) <= self.md5s.keys(), (
+            "South America Soybean data only exists for the following years: "
+            f"{list(self.md5s.keys())}."
+        )
+        assert (
+            set(classes) <= self.cmap.keys()
+        ), f"Only the following classes are valid: {list(self.cmap.keys())}."
+        assert 0 in classes, "Classes must include the background class: 0"
+
+
+        self.years = years
+        self.paths = paths
+        self.classes = classes
+        self.download = download
+        self.checksum = checksum
+        self.ordinal_map = torch.zeros(max(self.cmap.keys()) + 1, dtype=self.dtype)
+        self.ordinal_cmap = torch.zeros((len(self.classes), 4), dtype=torch.uint8)
+    
+        self._verify()
+
+        super().__init__(paths, crs, res, transforms=transforms, cache=cache)
+
+        for v, k in enumerate(self.classes):
+            self.ordinal_map[k] = v
+            self.ordinal_cmap[v] = torch.tensor(self.cmap[k])
+
+            
+    def __getitem__(self, query: BoundingBox) -> dict[str, Any]:
+        """Retrieve mask and metadata indexed by query.
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+        Returns:
+            sample of mask and metadata at that index
+        Raises:
+            IndexError: if query is not found in the index
+        """
+        sample = super().__getitem__(query)
+        sample["mask"] = self.ordinal_map[sample["mask"]]
+        return sample
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the extracted files already exist
+        if self.files:
+            return
+
+        # Check if the zip files have already been downloaded
+        exists = False
+
+        assert isinstance(self.paths, str)
+
+        #todo
+        pathname = os.path.join(self.paths, "**", self.zipfile_glob)
+        if glob.glob(pathname, recursive=True):
+            exists = True
+            self._extract()
+
+        if exists == True:
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.paths}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
```

Comment:
```suggestion
            raise DatasetNotFoundError(self)
```
Use new DatasetNotFoundError